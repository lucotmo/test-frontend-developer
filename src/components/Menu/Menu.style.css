.container{
  margin: 0;
  padding: 1rem;
  display: flex;
  justify-content: flex-end;
  align-items: center;
  list-style: none;
  background-color: #333;
}

.menu{
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  margin: 0;
  padding: 1rem;
  display: flex;
  flex-flow: column wrap;
  justify-content: space-between;
  transform: translateY(-120%);
  list-style: none;
  z-index: 10;
  transition: .5s all;
  background-color: #333;
}

.menu__link{
  margin: 0.1rem 0 0.1rem 0;
  padding: .5rem;
  border-radius: 4px;
  cursor: pointer;
  background-color: white;
}

.activelink {
  font-weight: 700;
  letter-spacing: 0.05rem;
  color: white;
  background-color: #d71209;
}

.menu__button{
  width: 32px;
  height: 28px;
  padding: 4px;
  right: 1rem;
  display: flex;
  flex-flow: column wrap;
  justify-content: space-between;
  align-items: center;
  border-radius: 4px;
  cursor: pointer;
  z-index: 20;
  background-color: #333;
}

.menu__line{
  width: 100%;
  height: 5px;
  background-color: white;
}

.menu--show{
  transform: translateY(40%);
}

@media (min-width: 920px) {
  .container{
    flex-flow: row nowrap;
  }
  .menu{
    flex-flow: row nowrap;
    justify-content: flex-start;
    transform: translateY(0%);
    z-index: initial;
  }

  .menu__link{
    margin: 0 .2rem 0 .2rem;
  }

  .menu__button{
    display: none;
  }

  .menu--show{
    transform: translateY(0%);
  }
}
