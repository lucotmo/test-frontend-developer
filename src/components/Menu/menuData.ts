export const menuData = [
  {
    id: '1',
    name: 'Avianca',
  },
  {
    id: '2',
    name: 'Vivair',
  }
];