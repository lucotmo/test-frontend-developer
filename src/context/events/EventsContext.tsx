import { createContext } from 'react';

interface EventsContextState {
  linkSelect: {
    id: string;
    name: string
  };
  changeLinkSelect: (id: string, name: string) => void
}

const EventsContext = createContext<EventsContextState | null>(null);

export default EventsContext;