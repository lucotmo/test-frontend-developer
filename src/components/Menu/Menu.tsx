import { useContext, useState } from 'react';

import EventsContext from '../../context/events/EventsContext';
const {menuData} = require('./menuData')

import './Menu.style.css'

const Menu = () => {
  const [showMenu, setShowMenu] = useState(false)
  
  // Extraer los valores del EventsContext
  const eventsContext = useContext(EventsContext);
  const {Â linkSelect, changeLinkSelect }: any = eventsContext;

  return (
    <nav className="container">
      <ul className={`menu ${showMenu && 'menu--show'}`} >
        {
          menuData.map((el: any) => {
            return <li 
              className={`menu__link ${el.id === linkSelect.id && 'activelink'}`} 
              key={el.id}
              onClick={() => {
                changeLinkSelect( `${el.id}`, `${el.name}` )
                setShowMenu(false)
              }}
            >{el.name}</li>
          })
        }
      </ul>
      <div className="menu__button" onClick={() => setShowMenu( !showMenu )}>
        <div className="menu__line"></div>
        <div className="menu__line"></div>
        <div className="menu__line"></div>
      </div>
    </nav>
  )
}

export default Menu
