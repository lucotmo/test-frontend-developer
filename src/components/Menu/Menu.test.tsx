import { mount } from 'enzyme';
import { menuData } from './menuData';
import Menu from './Menu';
import EventsContext from '../../context/events/EventsContext';

describe('Pruebas en <Menu />', () => {
  const changeLinkSelect = jest.fn();

  const wrapper = mount(
    <EventsContext.Provider value={{
      linkSelect: menuData[0],
      changeLinkSelect
    }}>
      <Menu />
    </EventsContext.Provider>
  )

  test('debe de mostrarse correctamente', () => {
    expect( wrapper ).toMatchSnapshot();
  });
})
