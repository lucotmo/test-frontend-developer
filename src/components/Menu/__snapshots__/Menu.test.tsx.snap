// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Menu /> debe de mostrarse correctamente 1`] = `
<nav
  className="container"
>
  <ul
    className="menu false"
  >
    <li
      className="menu__link activelink"
      onClick={[Function]}
    >
      Avianca
    </li>
    <li
      className="menu__link false"
      onClick={[Function]}
    >
      Vivair
    </li>
  </ul>
  <div
    className="menu__button"
    onClick={[Function]}
  >
    <div
      className="menu__line"
    />
    <div
      className="menu__line"
    />
    <div
      className="menu__line"
    />
  </div>
</nav>
`;
